Add dialog for picking up multiple items at a location

maetl opened the pull request:
Uses the existing `ItemDialog` rather than creating a new UI class.

A few small things that could be improved here, like encapsulating the `itemsAt(pos)[index]` access and dealing with a full inventory before popping up the dialog. It’s just the bare minimum changeset for now.

maetl commented on lib/src/ui/item_dialog.dart:
```diff
@@ -41,7 +48,7 @@ class ItemDialog extends Screen {
     }
 
-    if (keyCode == KeyCode.TAB) {
+    if (keyCode == KeyCode.TAB && canSwitchLocations) {
```
This now encapsulates the `allowedLocations.length` check. I kept this method in place because the check was used in two places.

maetl commented:
Updated as per suggestions (thanks!). Was able to get rid of the two special case boolean flags and let the `List<ItemLocation>` define the tab switching behaviour, so it feels a little less ad-hoc now.

Happy new year :smile:

munificent commented:
Sorry for the delay! One of my other hobby open source projects has been keeping me busy.

This patch is looking a lot nicer! I added a couple of minor comments. After those, I'll be happy to merge this in. :)

maetl commented:
> Sorry for the delay! One of my other hobby open source projects has been keeping me busy.

Sorry for the delay at this end too.

I’ve made the changes as noted. Added `final` and `const` modifiers to a couple of variable literals and fixed the linebreak/whitespace/style issues. Also simplified the code for rotating through `_command.allowedLocations` using the modulus operator, as you suggested.

munificent commented:
Groovy, thanks!
